configs/example/xpk.Dockerfile

```diff
@@ -0,0 +1,17 @@
+# Dockerfile for xpk examples in xpk_example_dag.py,
```

Comment:
I'm curious about the workflow on XPK-based runs - how is the docker image built? Is this handled by the infrastructure, or does the user build and tag the image manually?

Reply:
@jonb377, is there an existing image that your team normally uses with xpk?

When we built test images for all of our tests, it was just one more component that could break and fall out of sync with nightly.

Reply:
I believe the pattern used in MaxText is for developers to build their own baseline dependency image and iterate on top of that, so there's currently no pre-built nightly image. cc @Obliviour to fact check me.

I don't think it would be too much overhead to have MaxText CI pump out nightly images we could use in the tests, and the test scripts are already baked into the MaxText repo.

Reply:
One other thing I just remembered - if we use the same tag across executions, we need to ensure the XPK cluster has the latest version before running (the workloads won't pull from the server if the image exists locally). @Obliviour do you know if this is already handled for the XPK integration here?

Reply:
`I'm curious about the workflow on XPK-based runs - how is the docker image built? Is this handled by the infrastructure, or does the user build and tag the image manually?`

Thanks Jon for the question! Currently, the example image is manually built. @chandrasekhard2 has a project about docker image with Airflow, and it's in Q1 2024. Looking forward to his proposal/design.

If MaxText builds it's own images. I guess it can be stored in container repo, and pass it directly to xpk command?

Reply:
> I don't think it would be too much overhead to have MaxText CI pump out nightly images we could use in the tests, and the test scripts are already baked into the MaxText repo.

That would be helpful. I'd rather rely on an upstream image for testing. If we're building our own image, we're not really testing from the user's perspective.

Reply:
I've asked in the chat - they seem open to the idea of a nightly build. For now, we can just run the standard installation commands on top of a base image.